lib/shell.nix

```diff
@@ -182,7 +198,7 @@
 
       commonShell = pkgs.mkShell (cleanArgs
         // {
-          packages = (packages pkgs) ++ map commandToBin commandsList;
+          packages = (packages pkgs) ++ [pkgs.bashInteractive] ++ map commandToBin commandsList;
```

Comment:
Why you need this? `bashInteractive` will already be installed via `shell`.

Reply:
[This shell](https://github.com/terramagna/nixpkgs/pull/11/files#diff-cd78925af08c02668ad851141e7ed8fff395d5d6b8ea22af5c0d0029a2c213fdR130).

Reply:
Hmm. Yes, but the `SHELL` environment doesn't change what `bash` (the program) is being used. We also must ensure that the _interactive bash_ is provided.

Reply:
If you have `SHELL`, the user will never user `bash`, so it doesn't matter (you will need to export `SHELL` inside the shell though).

Reply:
The `SHELL` ensures that a program spawns the _correct_ shell (which might be `bashInteractive`). For example, Vim when one uses `:shell`.

However, `SHELL` isn't used when the user uses the `bash` command. Outside the Nix shell, when `bash` is invoked, the user gets an interactive version; which doesn't happen inside the Nix shell since, by default, stdenv provides the non-interactive version. I am trying to mirror the default behavior from outside the shell. You might argue that the user would rarely invoke `bash` directly, but I still can't see why the interactive version shouldn't be provided in an interactive context.

Reply:
I don't see a reason anyone would need an interactive `bash` but ok.